src/ui/ui_apps/ui_apps_mixin.js

```diff
@@ -0,0 +1,38 @@
+import { memoize } from 'lodash';
+
+import { UiApp } from './ui_app';
+
+export function uiAppsMixin(kbnServer, server) {
+
+  const { uiAppSpecs = [] } = kbnServer.uiExports;
+  const existingIds = new Set();
+
+  kbnServer.uiApps = uiAppSpecs.map(spec => {
```

Comment:
I'm also fine just getting rid of the memoize stuff, or indexing the list in a `uiAppsById` map instead of using `memoize()`, but I really dislike when lists of things aren't just arrays. Also, `Map#values()` returns an iterable, but not an array.

Reply:
:+1: